core/src/map.cpp

```diff
@@ -379,17 +379,23 @@ bool Map::update(float _dt) {
             markersNeedUpdate |= marker->isEasing();
         }
 
+        // FIXME: Set of labels needs to be updated when markers were added or removed.
+        // Otherwise Labels::m_labels may point to deleted Label instances.
+#if 0
         if (impl->view.changedOnLastUpdate() ||
-            impl->tileManager.hasTileSetChanged()) {
-
+            impl->tileManager.hasTileSetChanged() ||
+            markersNeedUpdate) {
```

Comment:
Maybe make sense to add a `dirty` state to `MarkerManager` and check that instead of running these updates every Tangram::update!